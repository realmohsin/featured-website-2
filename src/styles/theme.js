import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
