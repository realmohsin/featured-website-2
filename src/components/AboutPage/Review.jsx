import React, { useState } from 'react'
import { makeStyles, useTheme } from '@material-ui/core/styles'
import { Grid, Typography } from '@material-ui/core'
import { FaFacebook, FaStar } from 'react-icons/fa'
import { FcGoogle } from 'react-icons/fc'

const useStyles = makeStyles(theme => ({
  gridItem: {
    padding: '4.5rem 4rem',
    [theme.breakpoints.down('xs')]: {
      padding: '4rem 3.2rem'
    }
  },
  nameAndDate: {
    marginBottom: '0.6rem'
  },
  nameSpan: {
    color: theme.palette.primary.main
  },
  daysAgoSpan: {
    color: theme.palette.common.fadedTextColor
  },
  starsContainer: {
    display: 'flex',
    marginBottom: '1.8rem'
  },
  star: {
    marginRight: '0.3rem'
  },
  iconAndTextGrid: {
    display: 'flex'
  },
  text: {
    flex: 1,
    [theme.breakpoints.down('xs')]: {
      fontSize: '1.2rem'
    }
  },
  platformIcon: {
    marginTop: '1rem',
    marginRight: '1.5rem'
  },
  readMoreOrLess: {
    cursor: 'pointer',
    color: theme.palette.primary.light,
    textDecoration: 'underline',
    transition: '0.3s all',
    '&:hover': {
      color: theme.palette.primary.dark
    }
  }
}))

const Review = ({ review }) => {
  const theme = useTheme()
  const classes = useStyles()

  const [opened, setOpened] = useState(false)

  const toggleOpen = e => {
    setOpened(opened => !opened)
  }

  const handleKeyDown = e => {
    if (e.key === 'Enter' || e.key === ' ') {
      e.preventDefault()
      toggleOpen(e)
    }
  }

  return (
    <Grid item xs={12} sm={6} md={4} className={classes.gridItem}>
      <Typography variant='body1' className={classes.nameAndDate}>
        <span className={classes.nameSpan}>{review.name},</span> &nbsp;
        <span className={classes.daysAgoSpan}>{review.daysAgo}</span>
      </Typography>
      <div className={classes.starsContainer}>
        {Array(5)
          .fill(null)
          .map((_, index) => (
            <FaStar
              key={index}
              className={classes.star}
              size='1.2em'
              color={
                theme.palette.common[
                  review.type === 'facebook'
                    ? 'facebookColor'
                    : 'googleStarColor'
                ]
              }
            />
          ))}
      </div>
      <div className={classes.iconAndTextGrid}>
        <div>
          {review.type === 'facebook' ? (
            <FaFacebook
              color={theme.palette.common.facebookColor}
              size='2.4em'
              className={classes.platformIcon}
            />
          ) : (
            <FcGoogle size='2.4em' className={classes.platformIcon} />
          )}
        </div>
        <Typography variant='body2' className={classes.text}>
          {opened ? review.content : review.content.slice(0, 201)}
          {!opened && '...'} &nbsp;
          <span 
            className={classes.readMoreOrLess} 
            onClick={toggleOpen}
            onKeyDown={handleKeyDown}
            role="button"
            tabIndex={0}
          >
            Read {opened ? 'Less' : 'More'}
          </span>
        </Typography>
      </div>
    </Grid>
  )
}

export default Review
