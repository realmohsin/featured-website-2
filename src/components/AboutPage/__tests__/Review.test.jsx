import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import Review from "../Review";

// Mock react-icons
jest.mock("react-icons/fa", () => ({
  FaFacebook: ({ color, size, className }) => (
    <div data-testid="facebook-icon" style={{ color }} className={className}>
      FB
    </div>
  ),
  FaStar: ({ color, size, className }) => (
    <div data-testid="star-icon" style={{ color }} className={className}>
      â˜…
    </div>
  ),
}));

jest.mock("react-icons/fc", () => ({
  FcGoogle: ({ size, className }) => (
    <div data-testid="google-icon" className={className}>
      G
    </div>
  ),
}));

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", light: "#4A8B8F", dark: "#0A4A4D" },
    common: {
      fadedTextColor: "#999",
      facebookColor: "#1877F2",
      googleStarColor: "#FFC107",
    },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

const mockReview = {
  name: "John Doe",
  daysAgo: "2 days ago",
  type: "facebook",
  content: "This is a great law firm with excellent service. They helped me with my business legal needs and I couldn't be happier with the results. The team is professional and knowledgeable.",
};

const mockGoogleReview = {
  name: "Jane Smith",
  daysAgo: "1 week ago",
  type: "google",
  content: "Outstanding legal representation. Highly recommend their services for any business legal matters.",
};

describe("Review Component", () => {
  it("renders review with Facebook type", () => {
    const { container } = renderWithTheme(<Review review={mockReview} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders review with Google type", () => {
    const { container } = renderWithTheme(<Review review={mockGoogleReview} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays reviewer name and date", () => {
    renderWithTheme(<Review review={mockReview} />);
    expect(screen.getByText(/john doe/i)).toBeInTheDocument();
    expect(screen.getByText(/2 days ago/i)).toBeInTheDocument();
  });

  it("renders 5 star icons", () => {
    renderWithTheme(<Review review={mockReview} />);
    const stars = screen.getAllByTestId("star-icon");
    expect(stars).toHaveLength(5);
  });

  it("displays Facebook icon for Facebook reviews", () => {
    renderWithTheme(<Review review={mockReview} />);
    expect(screen.getByTestId("facebook-icon")).toBeInTheDocument();
    expect(screen.queryByTestId("google-icon")).not.toBeInTheDocument();
  });

  it("displays Google icon for Google reviews", () => {
    renderWithTheme(<Review review={mockGoogleReview} />);
    expect(screen.getByTestId("google-icon")).toBeInTheDocument();
    expect(screen.queryByTestId("facebook-icon")).not.toBeInTheDocument();
  });

  it("truncates long content initially", () => {
    renderWithTheme(<Review review={mockReview} />);
    expect(screen.getByText(/read more/i)).toBeInTheDocument();
    expect(screen.queryByText(/read less/i)).not.toBeInTheDocument();
  });

  it("expands content when Read More is clicked", () => {
    renderWithTheme(<Review review={mockReview} />);
    
    const readMoreButton = screen.getByText(/read more/i);
    fireEvent.click(readMoreButton);
    
    expect(screen.getByText(/read less/i)).toBeInTheDocument();
    expect(screen.queryByText(/read more/i)).not.toBeInTheDocument();
  });

  it("collapses content when Read Less is clicked", () => {
    renderWithTheme(<Review review={mockReview} />);
    
    const readMoreButton = screen.getByText(/read more/i);
    fireEvent.click(readMoreButton);
    
    const readLessButton = screen.getByText(/read less/i);
    fireEvent.click(readLessButton);
    
    expect(screen.getByText(/read more/i)).toBeInTheDocument();
    expect(screen.queryByText(/read less/i)).not.toBeInTheDocument();
  });

  it("handles keyboard interaction for Read More/Less", () => {
    renderWithTheme(<Review review={mockReview} />);
    
    const readMoreButton = screen.getByText(/read more/i);
    fireEvent.keyDown(readMoreButton, { key: "Enter" });
    
    expect(screen.getByText(/read less/i)).toBeInTheDocument();
  });
});
