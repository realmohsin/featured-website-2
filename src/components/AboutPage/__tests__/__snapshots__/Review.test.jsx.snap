// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Component renders review with Facebook type 1`] = `
<div
  class="MuiGrid-root makeStyles-gridItem-1 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
>
  <p
    class="MuiTypography-root makeStyles-nameAndDate-2 MuiTypography-body1"
  >
    <span
      class="makeStyles-nameSpan-3"
    >
      John Doe
      ,
    </span>
      
    <span
      class="makeStyles-daysAgoSpan-4"
    >
      2 days ago
    </span>
  </p>
  <div
    class="makeStyles-starsContainer-5"
  >
    <div
      class="makeStyles-star-6"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-6"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-6"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-6"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-6"
      data-testid="star-icon"
    >
      ★
    </div>
  </div>
  <div
    class="makeStyles-iconAndTextGrid-7"
  >
    <div>
      <div
        class="makeStyles-platformIcon-9"
        data-testid="facebook-icon"
      >
        FB
      </div>
    </div>
    <p
      class="MuiTypography-root makeStyles-text-8 MuiTypography-body2"
    >
      This is a great law firm with excellent service. They helped me with my business legal needs and I couldn't be happier with the results. The team is professional and knowledgeable.
      ...
        
      <span
        class="makeStyles-readMoreOrLess-10"
        role="button"
        tabindex="0"
      >
        Read 
        More
      </span>
    </p>
  </div>
</div>
`;

exports[`Review Component renders review with Google type 1`] = `
<div
  class="MuiGrid-root makeStyles-gridItem-11 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
>
  <p
    class="MuiTypography-root makeStyles-nameAndDate-12 MuiTypography-body1"
  >
    <span
      class="makeStyles-nameSpan-13"
    >
      Jane Smith
      ,
    </span>
      
    <span
      class="makeStyles-daysAgoSpan-14"
    >
      1 week ago
    </span>
  </p>
  <div
    class="makeStyles-starsContainer-15"
  >
    <div
      class="makeStyles-star-16"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-16"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-16"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-16"
      data-testid="star-icon"
    >
      ★
    </div>
    <div
      class="makeStyles-star-16"
      data-testid="star-icon"
    >
      ★
    </div>
  </div>
  <div
    class="makeStyles-iconAndTextGrid-17"
  >
    <div>
      <div
        class="makeStyles-platformIcon-19"
        data-testid="google-icon"
      >
        G
      </div>
    </div>
    <p
      class="MuiTypography-root makeStyles-text-18 MuiTypography-body2"
    >
      Outstanding legal representation. Highly recommend their services for any business legal matters.
      ...
        
      <span
        class="makeStyles-readMoreOrLess-20"
        role="button"
        tabindex="0"
      >
        Read 
        More
      </span>
    </p>
  </div>
</div>
`;
